src/fs.rs

```diff
@@ -558,16 +558,17 @@ impl Fs {
         matches!(self.lookup_node(path), Some(FsNode::File { .. }))
     }
 
-    #[allow(dead_code)]
     /// Get an iterator over the names of files/directories in a directory.
-    pub fn enumerate<P: AsRef<GuestPath>>(
-        &self,
-        path: P,
-    ) -> Result<impl Iterator<Item = &str>, ()> {
+    pub fn enumerate<P: AsRef<GuestPath>>(&self, path: P) -> Result<Vec<GuestPathBuf>, ()> {
         let Some(FsNode::Directory { children, .. }) = self.lookup_node(path.as_ref()) else {
             return Err(());
         };
-        Ok(children.keys().map(|name| name.as_str()))
+        let mut paths = Vec::new();
+        let iter = children.keys().map(|name| name.as_str());
+        for name in iter {
+            paths.push(GuestPathBuf::from(GuestPath::new(name)));
+        }
+        Ok(paths)
```

Comment:
Hmm, why did you change this? There's nothing wrong with it in principle, but having an iterator is usually better, and you could use `.map` to convert to strings in the calling function instead.

Reply:
Oh, was it to make it consistent with `enumerate_recursive`? That is sort of reasonable in its own way, but that function returns a `Vec<GuestPathBuf>` because it has to provide fully-qualified paths, while this function doesn't.

Reply:
I don't remember the exact reasoning as it was more than month ago :)
I saw that it was marked as [allow dead_code], so I took a liberty to modify the interface

Reply:
I changed signature back to the previous one, please check

Reply:
Looks good to me, thanks for changing it.